src/main/java/org/traccar/helper/MqttUtil.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.helper;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+
+import com.hivemq.client.mqtt.datatypes.MqttQos;
+import com.hivemq.client.mqtt.mqtt3.Mqtt3AsyncClient;
+import com.hivemq.client.mqtt.mqtt3.Mqtt3Client;
+import com.hivemq.client.mqtt.mqtt3.Mqtt3ClientBuilder;
+import com.hivemq.client.mqtt.mqtt3.message.auth.Mqtt3SimpleAuth;
+import com.hivemq.client.mqtt.mqtt3.message.publish.Mqtt3Publish;
+import com.nimbusds.oauth2.sdk.util.StringUtils;
+
+public final class MqttUtil {
+
+    private MqttUtil() {
+
+    }
+
+    public static Mqtt3AsyncClient createClient(final String url, final String willTopic) {
```

Comment:
What is `willTopic`? And why do we need to send "connected" and "disconnected" payload?

Reply:
This is standard a way to tell any other mqtt client that this application is connected or not.

We send "connected" when connected, and if the connection is lost the server will automatically send the "disconnected" payload to the topic. This is called last will.

An application that use this data have a way to know if this specific process is connected or not to the mqtt server.

in this case if admin configure the position forward to "/traccar/position" the code will send "connected" with retain flag to "/traccar/position/state" and when the traccar server stops and disconnect from mqtt server, the server will change this retained topic to "disconnected"

Reply:
Can you please provide a link to this standard.

Reply:
Here you can read more about this procedure : https://www.hivemq.com/blog/mqtt-essentials-part-9-last-will-and-testament/

Reply:
This doesn't look like a standard to me. Even the article implies that it's an optional thing.

Reply:
The article says "To summarize, the Last Will and Testament (LWT) feature in MQTT is crucial in ensuring efficient communication and maintaining system integrity in the event of unexpected client disconnections. By combining LWT with retained messages, developers can store and communicate client state on specific topics, providing valuable information to subscribed clients"

And in my environment I use this. This dont hurt (very low extra traffic) and gives extra context information.

Removing this will make more difficult to extend with my needs, in this case I will stay with my changes locally.

Reply:
OK. Let's leave the PR open in case anyone is interested in updating it later.

Reply:
Its 2 different things. Last will is used for health monitoring. I would not combine that in this PR. Last will behavior could be added later after this is merged?

Reply:
@tananaev if forwarding of connected and disconnected is disabled is this pr acceptable ?

Reply:
Yes, I think most other stuff is fine.

Reply:
Truly I don't see here anything wrong with LWT topic, it's used almost everywhere in same way as @ivanfmartinez implemented it. Through this topic all subscribers will know status of slender, if traccar suddenly dies, or disconnects from MQTT server, there's no way to know that.
@tananaev please accept this pr in full, i am also waiting for it.

Reply:
This PR won't be accepted as is. It does not make sense to implement a special case just for one forwarder.

Reply:
But it is, @tananaev , maybe you are seeing something more about it, but last will testament works like that, it set's retain flag for offline message, and keeps it updated with online message.. so that's how it's possible implement disconnected state in mqtt. Sorry if i don't understand something else...

Reply:
Correct, but I'm against including LWT in this PR.